src/main/java/org/opensearch/security/transport/SecurityInterceptor.java

```diff
@@ -268,14 +270,32 @@ && getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_ROL
         }
     }
 
+    private static final String USE_JDK_SERIALIZATION = "plugins.security.use_jdk_serialization";
+    private static final Version FIRST_CUSTOM_SERIALIZATION_SUPPORTED_OS_VERSION = Version.V_2_11_0;
+    private static final Version CUSTOM_SERIALIZATION_NO_LONGER_SUPPORTED_OS_VERSION = Version.V_2_14_0;
+
+    private SerializationFormat shouldUseJdkSerialization(final Connection connection) {
+        var version = connection.getVersion();
+        if (version.after(FIRST_CUSTOM_SERIALIZATION_SUPPORTED_OS_VERSION)
+            && version.before(CUSTOM_SERIALIZATION_NO_LONGER_SUPPORTED_OS_VERSION)) {
```

Comment:
@cwperks What do you think of this as the mitigation?  Being that the cost of building a custom serialization solution that also handles deduplication and compression is likely out of scope for the project in the long run I think its safe route to take.

So look as this looks good to you, then I'll start building out more test validation for these changes

Reply:
This looks ok. Let's say the de-dupe is fixed in a future minor version. Would implementing a bwc change to the new serialization format mean adding an `else if` block here?

Reply:
Yup, it could.